server/service/integration_mdm_setup_experience_test.go

```diff
@@ -3943,9 +3943,387 @@ func (s *integrationMDMTestSuite) TestAndroidAppConfiguration() {
 		Configuration: json.RawMessage(`{"managedConfiguration": 3}`),
 	}, http.StatusOK, &patchAppResp)
 
-	s.runWorkerUntilDone()
+	s.runWorkerUntilDoneWithChecks(true)
 
 	require.Len(t, patchAppsPolicies, 0)
+
+	// patch with a different config just to trigger the worker
+	s.DoJSON("PATCH", fmt.Sprintf("/api/latest/fleet/software/titles/%d/app_store_app", app3TitleID), &updateAppStoreAppRequest{
+		TeamID:        nil,
+		Configuration: json.RawMessage(`{}`),
+	}, http.StatusOK, &patchAppResp)
+
+	// delete directly the config from the DB (it seems like to clear the config from
+	// the API, an empty object needs to be passed, but that won't clear the config,
+	// it just won't change any managedConfig/widgets - to really clear the config
+	// from the API, the user would have to send something like:
+	// {
+	//   "managedConfiguration": null,
+	//   "workProfileWidgets": "WORK_PROFILE_WIDGETS_UNSPECIFIED"
+	// }
+	//
+	// Is that how we want it to work?
```

Comment:
TBD (again, probably requires product input) how we want to make it possible to clear any app config and revert back to defaults? If they pass an empty object, it will just not set anything for `managedConfiguration` and `workProfileWidgets`, resulting in Google keeping the current values as-is (the patch policy apps is patch semantics as its name implies).

Also note that there is currently no way to _delete_ an existing app config - in particular, it does not get deleted/cleared when the corresponding team gets deleted. It only gets deleted if the corresponding VPP app gets deleted for a team.